source/common/upstream/cluster_manager_impl.cc

```diff
@@ -604,26 +604,22 @@ void ClusterManagerImpl::ThreadLocalClusterManagerImpl::drainConnPools(
 
 void ClusterManagerImpl::ThreadLocalClusterManagerImpl::drainConnPools(
     HostSharedPtr old_host, ConnPoolsContainer& container) {
-  for (const Http::ConnectionPool::InstancePtr& pool : container.pools_) {
-    if (pool) {
-      container.drains_remaining_++;
-    }
+  std::vector<Http::ConnectionPool::Instance*> to_be_drained;
```

Comment:
Why do we need to copy in temporary first?

Reply:
We did when I initially made this change couple of months ago, but now I see there is the explicit break out in the case the container was nuked. I'll revert this and retest.